### PR TITLE
Contained Radio & Checkbox

### PR DESCRIPTION
**Pull request by annekao**
https://www.figma.com/file/BrXOVNTglDWg03DL7ZZeW1/Teamshares-UI?t=ymYY1RK2fUGLGvQM-0

slack discussion on implementation: https://teamsharesinc.slack.com/archives/C04KR6HQ5L7/p1679584015946389

Screenshot 2023-03-24 at 11 17 11 AM
Screenshot 2023-03-24 at 11 16 55 AM

### CODE REVIEW

File: src/components/checkbox/checkbox.ts [hunk 1]
```diff
@@ -231,8 +241,10 @@ export default class SlCheckbox extends ShoelaceElement implements ShoelaceFormC
             : ''}
         </span>
 
-        <div part="label" class="checkbox__label">
-          <slot></slot>
+        <div class="checkbox__label-description-container">
+          <slot part="label" class="checkbox__label"></slot>
+          <div class="checkbox__description-block"></div>
```
Comment (annekao) [inline src/components/checkbox/checkbox.ts:246]: this is to put space between the label and description. it's _not ideal_, but this section has a line-height (in order for it to lineup with the checkbox/radio) which does not allow for a margin 😭

Comment (CrookedGrin) [reply src/components/checkbox/checkbox.ts:246]: Yeah, styling checkboxes has always been a PITA

**Review by CrookedGrin — CHANGES_REQUESTED** [review]
This looks great! I checked (ha ha) on the other usages of checkbox, such as in `<sl-menu-item>` and in forms, and it all looks good. The one missing thing is to bump the version (still showing 1.0.6, would maybe call this 1.1.0 since it changed the rendering of the checkbox) and add a changelog entry. Then once we merge we can cut another release.

**Review by CrookedGrin — APPROVED** [review]
Awesome, thanks!

**Pull request merged**